sky/utils/schemas.py

```diff
@@ -85,6 +85,19 @@ def _get_single_resources_schema():
             'zone': {
                 'type': 'string',
             },
+            'infra': {
+                'type': 'string',
+                'description':
+                    ('Infrastructure specification in format: '
+                     'cloud[/region[/zone]]. Use "*" as a wildcard.'),
+                # Create a pattern validator that uses a big regex to match all
+                # valid formats. This allows us to maintain JSON Schema
+                # validation while supporting all formats
+                'pattern':
+                    ('^(?:(?i:(' + '|'.join(list(service_catalog.ALL_CLOUDS)) +
+                     '))(?:/[^/]+(?:/[^/]+)?)?|\\*(?:/[^/]+(?:/[^/]+)?|/\\*'
+                     '(?:/[^/]+)?)?|(?i:k8s|kubernetes)/.+)$')
```

Comment:
Human interpretation for future reference:
- `cloud/x?/x?` format:
  - Cloud name
  - Optionally /region
  - Optionally /zone (must have /region)
  - (Note: technically I believe this could be simplified e.g. `(?:/[^/]){0,2}` but it's equivalent)
- `*/x/x?`
  - */region - region is required
  - Optionally /zone
- k8s/anything or kubernetes/anything
  - context name can include slashes
  - this will need to be updated to include ssh

Reply:
Review comment: `*/*` matches which seems unintentional.
Either way, the `|/\*'(?:/[^/]+)?` after `*` cloud seems completely redundant to the simpler `/[^/]`... match before

Reply:
Updated the comments and fixed the schemas : )